### ISSUE TITLE
Add LuaState as first parameter in C# function

### ISSUE THREAD

**Issue by willzhanglala** [NLua/NLua#229]
It's a feature request actually.
In some case, we cannot get a global LuaState, (maybe don't have access, or it's just a local variable).  So that I hope the WrapperCall can try to find a function that first parameter is a LuaState, and then pass the state to this function.
```
class SomeClass
{
	public static LuaTable json_decode(KeraLua.LuaState state, string json)
	{			
		LuaLib.LuaNewTable(state);
		// ...  to parse json string
		Lua lua = new Lua(state);
		return lua.Pop() as LuaTable;			
	}
}
```
So that in lua
I can call 
```
SomeClass.json_decode( aString )
```
will map to 
```
SomeClass::json_decode(LuaState state, string json)
```

**Comment by willzhanglala** [comment]
Or maybe we can use Lua as the first parameter

**Comment by CWolfs** [comment]
This would be useful for me too.

**Comment by CWolfs** [comment]
@willzhanglala Just as a FYI, I made a dirty hack on my version to get it to work since I run with the NLua source in my game.

I added the following lines under https://github.com/NLua/NLua/blob/master/Core/NLua/Metatables.cs#L1259

`
if (currentNetParam.ParameterType == typeof(LuaState)) {
    paramList.Add(luaState);
    continue;
}
`
I know it's hacky but it works for me. Now, if you put 'LuaState' type in any of your methods it will autopopulate it for you. I put it at the end of the methods which need it for me.
